### PR TITLE
e2e_upgrades: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by Songjoy**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/upgrades/apps/deployments.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 	"k8s.io/kubernetes/test/e2e/upgrades"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
 	imageutils "k8s.io/kubernetes/test/utils/image"
```
Comment (carlory) [inline test/e2e/upgrades/apps/deployments.go:36]: please leave an empty line.

File: test/e2e/upgrades/apps/deployments.go [hunk 2]
```diff
@@ -132,27 +133,27 @@ func (t *DeploymentUpgradeTest) Test(ctx context.Context, f *framework.Framework
...
 	case rss[1].UID:
-		framework.ExpectEqual(rss[0].UID, t.newRSUID)
+		gomega.Expect(rss[0].UID).To(gomega.Equal(t.newRSUID))
 	default:
 		framework.ExpectNoError(fmt.Errorf("new replicasets are created during upgrade of deployment %q", deploymentName))
 	}
 
 	ginkgo.By(fmt.Sprintf("Verifying revision of the deployment %q is still 2", deploymentName))
-	framework.ExpectEqual(deployment.Annotations[deploymentutil.RevisionAnnotation], "2")
+	gomega.Expect(deployment.Annotations[deploymentutil.RevisionAnnotation]).To(gomega.Equal("2"))
```
Comment (carlory) [inline test/e2e/upgrades/apps/deployments.go:156]: please use HaveKeyWithValue instead.

**Review by pohly — APPROVED** [review]
/approve

**Pull request merged**